components/menu/index.css

```diff
@@ -291,17 +291,46 @@
 	}
 }
 
-.spectrum-Menu-itemIcon {
+.spectrum-Menu-itemIcon.spectrum-Icon {
 	fill: var(--highcontrast-menu-item-color-default, var(--mod-menu-item-label-icon-color-default, var(--spectrum-menu-item-label-icon-color-default)));
 	color: var(--highcontrast-menu-item-color-default, var(--mod-menu-item-label-icon-color-default, var(--spectrum-menu-item-label-icon-color-default)));
 }
 
-.spectrum-Menu-checkmark {
-	align-self: center;
+.spectrum-Menu-checkmark.spectrum-Icon {
 	display: var(--mod-menu-checkmark-display, var(--spectrum-menu-checkmark-display));
+	block-size: var(--mod-menu-item-checkmark-height, var(--spectrum-menu-item-checkmark-height));
+	inline-size: var(--mod-menu-item-checkmark-width, var(--spectrum-menu-item-checkmark-width));
+
 	fill: var(--highcontrast-menu-checkmark-icon-color-default, var(--mod-menu-checkmark-icon-color-default, var(--spectrum-menu-checkmark-icon-color-default)));
 	color: var(--highcontrast-menu-checkmark-icon-color-default, var(--mod-menu-checkmark-icon-color-default, var(--spectrum-menu-checkmark-icon-color-default)));
 	opacity: 1;
+
+	margin-block-start: calc(var(--mod-menu-item-top-to-checkmark, var(--spectrum-menu-item-top-to-checkmark)) - var(--mod-menu-item-top-edge-to-text, var(--spectrum-menu-item-top-edge-to-text)));
+
+	/* Checkmarks have text or an icon directly next to them. */
+	margin-inline-end: var(--mod-menu-item-text-to-control, var(--spectrum-menu-item-text-to-control));
+}
+
+.spectrum-Menu-chevron.spectrum-Icon {
+	block-size: var(--spectrum-menu-item-checkmark-height);
+	inline-size: var(--spectrum-menu-item-checkmark-width);
+
+	/* Chevrons (aka collapsibles) have text or an icon directly next to them. */
+	margin-inline-end: var(--mod-menu-item-text-to-control, var(--spectrum-menu-item-text-to-control));
+}
+
+.spectrum-Menu-chevron:dir(rtl) {
+	transform: rotate(-180deg);
+}
+
+.spectrum-Menu-itemIcon:not(.spectrum-Menu-chevron, .spectrum-Menu-checkmark) {
+	block-size: var(--mod-menu-item-icon-height, var(--spectrum-menu-item-icon-height));
+	inline-size: var(--mod-menu-item-icon-width, var(--spectrum-menu-item-icon-width));
+}
+
+.spectrum-Menu-itemIcon--workflowIcon {
+	/* Always provide space at the end of a workflow icon. */
+	margin-inline-end: var(--mod-menu-item-label-text-to-visual, var(--spectrum-menu-item-label-text-to-visual));
```

Comment:
All these additions are moved up from below. The only changes should be the addition of `.spectrum-Icon` to a couple selectors.